MonacoEditorComponent/Monaco/Languages/CompletionItem.cs

```diff
@@ -1,51 +1,129 @@
+        /// this completion.
+        /// is used.
+        /// </summary>
+        [JsonProperty("insertText")]
+        public string InsertText { get; set; }
 
+        /// <summary>
+        /// Addition rules (as bitmask) that should be applied when inserting
+        /// this completion.
+        /// </summary>
+        [JsonProperty("insertTextRules", NullValueHandling = NullValueHandling.Ignore)]
+        public CompletionItemInsertTextRule? InsertTextRules { get; set; }
+
+        /// <summary>
+        /// The kind of this completion item. Based on the kind
+        /// an icon is chosen by the editor.
+        /// </summary>
         [JsonProperty("kind")]
         public CompletionItemKind Kind { get; set; }
 
+        /// <summary>
+        /// The label of this completion item. By default
+        /// this is also the text that is inserted when selecting
+        /// this completion.
+        /// </summary>
         [JsonProperty("label")]
         public string Label { get; set; }
 
-        [JsonProperty("range", NullValueHandling = NullValueHandling.Ignore)]
-        public IRange Range { get; set; }
+        /// <summary>
+        /// Select this item when showing. *Note* that only one completion item can be selected and
+        /// that the editor decides which item that is. The rule is that the *first* item of those
+        /// that match best is selected.
+        /// </summary>
+        [JsonProperty("preselect", NullValueHandling = NullValueHandling.Ignore)]
+        public bool? Preselect { get; set; }
+
+        /// <summary>
+        /// A range of text that should be replaced by this completion item.
+        ///
+        /// Defaults to a range from the start of the [current
+        /// word](#TextDocument.getWordRangeAtPosition) to the
+        /// current position.
+        ///
+        /// *Note:* The range must be a [single line](#Range.isSingleLine) and it must
+        /// [contain](#Range.contains) the position at which completion has been
+        /// [requested](#CompletionItemProvider.provideCompletionItems).
+        /// </summary>
+        [JsonProperty("range")]
+        public Range Range { get; set; }
 
+        /// <summary>
+        /// A string that should be used when comparing this item
+        /// with other items. When `falsy` the [label](#CompletionItem.label)
+        /// is used.
+        /// </summary>
         [JsonProperty("sortText", NullValueHandling = NullValueHandling.Ignore)]
         public string SortText { get; set; }
 
-        public CompletionItem(string label, CompletionItemKind kind)
+        /// <summary>
+        /// A modifier to the `kind` which affect how the item
+        /// is rendered, e.g. Deprecated is rendered with a strikeout
+        /// </summary>
+        [JsonProperty("tags", NullValueHandling = NullValueHandling.Ignore)]
+        public MarkerTag[] Tags { get; set; }
+
+        public CompletionItem(string label, string insertText, CompletionItemKind kind)
```

Comment:
We should call this out on the breaking changes too, eh?

Reply:
Yes, I'll add it to changelog later